### PR TITLE
Fix edit command error msg

### LINKED ISSUE #210: Commands: Edit command should show the part in the format that is wrong and not just a general msg.
Edit command currently shows this if I type in `edit -h key:value` (note that this is not allowed as a header).  
`Invalid command format! 
edit: Edits the details of an existing API endpoint identified using it's displayed index from the API endpoint list. Existing values will be overwritten by the input values.
Parameters: INDEX (must be a positive integer) [-x METHOD] [-u ADDRESS] [-d DATA] [-h HEADER]...
[-t TAG]...
Example: edit 1 -u wall street `
Should show just the error area.

### PR DESCRIPTION
**Pull request by JulietTeoh**
and made sure other command error msgs were properly formatted

The error is that if there is a space in PREFIX_METHOD and other prefixs, it will messes up how the index is parsed, and also messes up some functionalities like "edit 1 -t " in order to take away existing tags

closes #210

### CODE REVIEW

**Comment by tjtanjin** [comment]
I did a quick skim of the files and it seems that the whitespace removed from the likes of `PREFIX_METHOD` are instead being introduced in the strings they are being concatenated with. Am I overlooking anything?

**Comment by JulietTeoh** [comment]
> I did a quick skim of the files and it seems that the whitespace removed from the likes of `PREFIX_METHOD` are instead being introduced in the strings they are being concatenated with. Am I overlooking anything?

Hi the only thing in the PR that affects code is the prefixs being changed in CliSyntax.java. For the add, edit and run, I just made the error command messages more readable, for example: from 
"add -xget -usomeurl -hsomeheader" to 
"add -x get -u someurl -h someheader"

**Comment by tjtanjin** [comment]
> > I did a quick skim of the files and it seems that the whitespace removed from the likes of `PREFIX_METHOD` are instead being introduced in the strings they are being concatenated with. Am I overlooking anything?
> 
> Hi the only thing in the PR that affects code is the prefixs being changed in CliSyntax.java. For the add, edit and run, I just made the error command messages more readable, for example: from
> "add -xget -usomeurl -hsomeheader" to
> "add -x get -u someurl -h someheader"

Okie @ong6 introduced the whitespace in `PREFIX_METHOD` in this [commit](https://github.com/AY2021S2-CS2103T-T12-4/tp/commit/750905738da4644176a8ed3d54a5afdcd73ac6e9) 5 days ago so I think's best for him to do the review.

**Comment by JulietTeoh** [comment]
> Okie @ong6 introduced the whitespace in `PREFIX_METHOD` in this [commit](https://github.com/AY2021S2-CS2103T-T12-4/tp/commit/750905738da4644176a8ed3d54a5afdcd73ac6e9) 5 days ago so I think's best for him to do the review.

@ong6 so heres a more in depth view of whats wrong with PREFIX_METHOD = "-x "

using an example of "edit 1 -x "
when passed into the parseIndex method of ParserUtil.java, it calls 
string trimmedIndex = oneBasedIndex.trim(); which trims the string to "edit 1 -x".

then because PREFIX_METHOD is "-x " (with the space), StringUtil.isNonZeroUnsignedInteger(trimmedIndex) will be true, because trimmedIndex will be equal to "1 -x", which leads an exception of the EditCommand.MESSAGE_INVALID_INDEX being thrown, instead of a more specific error message from the parseMethod method.

Also, if PREFIX_METHOD = "-x ", doing "edit 1 -t " or "edit 1 -h " does not work to remove tags/header because of the same reasons as stated above.

**Comment by ong6** [comment]
> > Okie @ong6 introduced the whitespace in `PREFIX_METHOD` in this [commit](https://github.com/AY2021S2-CS2103T-T12-4/tp/commit/750905738da4644176a8ed3d54a5afdcd73ac6e9) 5 days ago so I think's best for him to do the review.
> 
> @ong6 so heres a more in depth view of whats wrong with PREFIX_METHOD = "-x "
> 
> using an example of "edit 1 -x "
> when passed into the parseIndex method of ParserUtil.java, it calls
> string trimmedIndex = oneBasedIndex.trim(); which trims the string to "edit 1 -x".
> 
> then because PREFIX_METHOD is "-x " (with the space), StringUtil.isNonZeroUnsignedInteger(trimmedIndex) will be true, because trimmedIndex will be equal to "1 -x", which leads an exception of the EditCommand.MESSAGE_INVALID_INDEX being thrown, instead of a more specific error message from the parseMethod method.
> 
> Also, if PREFIX_METHOD = "-x ", doing "edit 1 -t " or "edit 1 -h " does not work to remove tags/header because of the same reasons as stated above.
> 
> TLDR: there should be no space in all prefixs, unless you want to change a ton of other codde

Nice spot! 

But then this would allow users to do edit 1 -tdog (without space), I guess it would be possible to remedy this somewhere else, but if I'm not wrong, the original AB3 code came with a "/t " (with space) so when changing to "-t " I kept the space as well. 

However, after checking out your commit, I think that the problems that the space brings are substantial so this is a good change overall!

**Review by ong6 — APPROVED** [review]
LGTM
